src/coreclr/vm/jitinterface.cpp

```diff
@@ -11605,6 +11605,45 @@ InfoAccessType CEEJitInfo::emptyStringLiteral(void ** ppValue)
     return result;
 }
 
+// Finds FieldDesc* that corresponds to the given offset in the given type and returns its value if it's nullptr or NonGC heap object
+bool CEEInfo::TryGetFieldObjectHandle(size_t baseAddr, MethodTable* structTypeMT, unsigned offset, CORINFO_OBJECT_HANDLE* handle)
```

Comment:
Does the JIT have the handle of the field that it is interested in? If yes, would it be better for the JIT to pass it in instead of VM trying to do this reverse mapping?

Reply:
> Does the JIT have the handle of the member field that it is interested in?

Or type handle of the member field that it is interested in.

Reply:
> Does the JIT have the handle of the member field that it is interested in? If yes, would it be better for the JIT to pass it in instead of VM trying to do this reverse mapping?

Currently, we don't require that so users of `getStaticFieldContent` can do whatever they want e.g. `Unsafe.Read<long>` on a struct. Or getting a char at N-index in a string object (obvisously, we don't have any field handle for that). Only ref fields are special as we want to prevent JIT from accidently reading direct addresses or parts of them. 

I've done ref-field support only for `OperationSystem` struct to be able to fold e.g. `RuntimeInformation.IsOSPlatform(OSPlatform.Windows);` in JIT - including _optional_ sub-field handle is a bit of work as currently we just have a general "calculate the final offset from a graph of GT_ADD" algorithm to accumulate the final offset and pass it to this API, but instead we'll have to detect a specific CNS_INT with the given FieldSeq. I can do this, it's just that I believe currently there is only one use of it - the `OperationSystem` and this PR didn't introduce that loop - it just moved it to a separate function.

Reply:
> Or type handle of the member field that it is interested in.

Will type handle help anyhow to find the right field? Presumably a struct may contain multiple fields of the same type? (and well it won't help for e.g.)

```cs
static ReadOnlySpan<byte> RVA => new byte[] { 0x00, 0x00, 0x01, 0x00 };

int Foo() => MemoryMarshal.Read<int>(RVA);
```
this currently is lowered to:
```asm
; Assembly listing for method Program:Foo():int (Tier1)
       mov      eax, 0x10000
       ret      
; Total bytes of code 6
```

Reply:
Is this function even necessary? We've verified in in `getStaticFieldContent` that the JIT is requesting a pointer sized buffer and that there is a GC pointer at this offset. Can't we just check there for `ptrs[i] == TYPE_GC_REF` and then do the remaining frozen segment checks if that passes? That will also support e.g. `InlineArray` containing object references to frozen segment.

Reply:
In other words, the loop there can copy the data over to the JIT as long as `ptrs[i] == TYPE_GC_NONE || (ptrs[i] == TYPE_GC_REF && IsFrozen(i))`, can't it?

Reply:
Ouch, not sure why I didn't consider this - done